src/DetailsView/components/load-assessment-button.tsx

```diff
@@ -3,31 +3,125 @@
 
 import { AssessmentDataParser } from 'common/assessment-data-parser';
 import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
+import { Tab } from 'common/itab';
+import {
+    AssessmentStoreData,
+    PersistedTabInfo,
+} from 'common/types/store-data/assessment-result-data';
+import { TabStoreData } from 'common/types/store-data/tab-store-data';
+import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
+import { UrlParser } from 'common/url-parser';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
-import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
+import { LoadAssessmentDialog } from 'DetailsView/components/load-assessment-dialog';
 import * as React from 'react';
 
 export type LoadAssessmentButtonDeps = {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
     assessmentDataParser: AssessmentDataParser;
-    loadAssessmentHelper: LoadAssessmentHelper;
+    urlParser: UrlParser;
+    fileReader: FileReader;
+    document: Document;
 };
 export interface LoadAssessmentButtonProps {
     deps: LoadAssessmentButtonDeps;
+    tabStoreData: TabStoreData;
+    assessmentStoreData: AssessmentStoreData;
+}
+export interface LoadAssessmentButtonState {
+    newTab: Tab;
+    prevTab: PersistedTabInfo;
+    loadedAssessmentData?: VersionedAssessmentData;
+    show: boolean;
 }
 
 export const loadAssessmentButtonAutomationId = 'load-assessment-button';
 
-export class LoadAssessmentButton extends React.Component<LoadAssessmentButtonProps> {
+export class LoadAssessmentButton extends React.Component<
+    LoadAssessmentButtonProps,
+    LoadAssessmentButtonState
+> {
+    public constructor(props) {
+        super(props);
+        this.state = {
+            loadedAssessmentData: null,
+            newTab: this.getNewTab(),
+            prevTab: props.assessmentStoreData.persistedTabInfo,
+            show: false,
+        };
+    }
+
+    private toggleLoadDialog() {
+        this.setState(prevState => ({ show: !prevState.show }));
+    }
+
+    private getNewTab(): Tab {
+        if (this.state !== undefined && this.state.loadedAssessmentData != null) {
+            return {
+                id: this.state.loadedAssessmentData.assessmentData.persistedTabInfo.id,
+                url: this.state.loadedAssessmentData.assessmentData.persistedTabInfo.url,
+                title: this.state.loadedAssessmentData.assessmentData.persistedTabInfo.title,
+            };
+        }
+
+        return {
+            id: this.props.tabStoreData.id,
+            url: this.props.tabStoreData.url,
+            title: this.props.tabStoreData.title,
+        };
+    }
+
+    public getAssessmentForLoad() {
```

Comment:
It might make sense to move this back to the separate method/class and instead add parameters for what this requires from the button (state and function that toggles load dialog).